x-pack/plugins/endpoint/server/routes/metadata/index.ts

```diff
@@ -138,9 +161,32 @@ function mapToHostResultList(
   }
 }
 
-function enrichHostMetadata(hostMetadata: HostMetadata): HostInfo {
+async function enrichHostMetadata(
+  hostMetadata: HostMetadata,
+  metadataRequestContext: MetadataRequestContext
+): Promise<HostInfo> {
+  let hostStatus = HostStatus.ERROR;
+  try {
+    const status = await metadataRequestContext.endpointAppContext.agentService.getAgentStatus(
+      metadataRequestContext.requestHandlerContext,
+      hostMetadata.elastic.agent.id
+    );
```

Comment:
```suggestion
    const { endpointAppContext, requestHandlerContext } = metadataRequestContext;
    const { getAgent, getAgentStatus } = endpointAppContext.agentService;
    const agent = await getAgent(requestHandlerContext.core.savedObjects.client, hostMetadata.elastic.agent.id);
    const status = await getAgentStatus(agent);
```

Reply:
I won't be able to try this locally until Monday afternoon, but I think this will work. If so, we can avoid the new `getAgentStatus` this PR adds in Ingest Manager.

Reply:
Can we talk about this. I would think a wholesale service will be better that performs the functionality will be better. And this does not isolate from changes to how agent status is derived. With this approach we have too much knowledge of the internal working of Ingest.

Reply:
Sure, we can talk anytime. I'm available most days from 1pm-5pm ET.

Before then, I'll try to summary my understanding & suggestions.

As I understand it, this PR [adds an Agent Service](https://github.com/elastic/kibana/pull/63921/files#diff-08697ba2f61ae23a7767744db13dfc19R11) but Ingest Manager [already has one ](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/index.ts)

It also [adds a function `getAgentStatus`](https://github.com/elastic/kibana/pull/63921/files#diff-08697ba2f61ae23a7767744db13dfc19R13) which a) is a duplicate of [an existing function](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/status.ts#L20) and b) uses the existing services functions from Ingest.

I'd like to avoid both those issues and think we can do that by moving that function into [`services/agents/status`](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/status.ts), changing its name to something like `getAgentStatusById`, and have it accept a saved object client and string id. That's in line with lots of our other service functions.

That way, all you're really changing at this call site is the function name and the first argument. It's still only calling one function and Ingest is the one in control of that logic.

```diff
-    const status = await metadataRequestContext.endpointAppContext.agentService.getAgentStatus(
-      metadataRequestContext.requestHandlerContext,
-      hostMetadata.elastic.agent.id
-    );
+    const status = await metadataRequestContext.endpointAppContext.agentService.getAgentStatusById(
+      metadataRequestContext.requestHandlerContext.core.savedObjects.client,
+      hostMetadata.elastic.agent.id
+    );
```

Hope that helps. Reach out any time to talk more.